actions/file_actions.ts

```diff
@@ -127,6 +136,43 @@ export function uploadFile({file, name, type, rootId, channelId, clientId, onPro
 
         xhr.send(formData);
 
+        dispatch({
+            type: FileTypes.START_UPLOADING_FILE,
+            clientId,
+            data: xhr,
+        });
+
         return xhr;
     };
 }
+
+export function cancelUploadingFile(clientId: string) {
```

Comment:
could you please add some comments explaining this piece of code to help future contributors better understand it? Some of the things here are rather cryptic (e.g. why are we checking for `pendingPostEntry[1].length !== 1`?)